packages/flutter/lib/src/painting/text_linker.dart

```diff
@@ -0,0 +1,424 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+
+import 'inline_span.dart';
+import 'text_span.dart';
+
+/// Signature for a function that builds an [InlineSpan] link.
+///
+/// The link displays [displayString] and links to [linkString] when tapped.
+/// These are distinct because sometimes a link may be split across multiple
+/// [TextSpan]s.
```

Comment:
Maybe add an example with a list of textspan and how would each displayString and linkString is called.

Reply:
Ok I added text_linker.1.dart and included some comments explaining the idea a bit more.

Reply:
Oh I originally meant adding here in the doc, something like
```
If you have TextSpans like this.
[TextSpan(text: 'http://'), TextSpan(text: 'google.com'), TextSpan(text: '/')]

This builder will call three time with
1st: displayString = 'http://', linkString = 'http://google.com/'
2nd: displayString = 'google.com', linkString = 'http://google.com/'
2nd: displayString = '/', linkString = 'http://google.com/'

```

Reply:
That's useful too, done.